akka-remote/src/main/java/akka/remote/artery/AeronErrorLog.java

```diff
@@ -39,24 +36,26 @@
  */
 public class AeronErrorLog
 {
-    private final File cncFile;
     final MappedByteBuffer cncByteBuffer;
-    final DirectBuffer cncMetaDataBuffer;
-    final int cncVersion;
     final AtomicBuffer buffer;
 
+    @Deprecated
     public AeronErrorLog(File cncFile)
     {
-      this.cncFile = cncFile;
+      this(cncFile, NoLogging.getInstance());
+    }
+
+    public AeronErrorLog(File cncFile, LoggingAdapter log)
+    {
       cncByteBuffer = IoUtil.mapExistingFile(cncFile, "cnc");
-      cncMetaDataBuffer = CncFileDescriptor.createMetaDataBuffer(cncByteBuffer);
-      cncVersion = cncMetaDataBuffer.getInt(CncFileDescriptor.cncVersionOffset(0));
+      final DirectBuffer cncMetaDataBuffer = CncFileDescriptor.createMetaDataBuffer(cncByteBuffer);
```

Comment:
IMHO using the final keyword is a good thing, think of it as `val` vs `var`, it will make the JIT work less at the end.
At least I always use `final` modifier where possible except for method arguments which look ugly.